lib/reek/smells/boolean_parameter.rb

```diff
@@ -1,4 +1,5 @@
 require_relative 'smell_detector'
+require_relative 'smell_warning'
```

Comment:
> The arguments from above are valid for sure. But putting this require in EVERY smell just seems wrong.

Well, that’s because every smell is coupled to both `SmellDetector` and `SmellWarning`. :)

>  We should put it in the SmellDetector. Practical usage over architectural correctness.:)

Oh definitely. In my pet projects I’m starting to lean towards ‘this is required 20 times across the project, so it’s a core concept of the project, let’s move the require to `lib/foo.rb`’).